client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx

```diff
@@ -0,0 +1,157 @@
+    },
+    [searchText],
+    500
+  );
+
+  const handleDataUpdate = (selectedResult: Record<string, any> | null) => {
+    if (selectedResult === null) return;
+    if (!saveFields) handleChange(path, selectedResult);
+    else {
+      const newObj: Record<string, any> = {};
+      saveFields?.forEach(
+        field => (newObj[field] = selectedResult[field] ?? null)
+      );
+      handleChange(path, newObj);
+    }
+    setEditMode(false);
+  };
+
+  const getNoOptionsText = () => {
+    switch (true) {
+      case loading:
+        return t('control.search.searching-label');
+      case searchText.length < MIN_CHARS:
+        return t('control.search.below-min-chars', { minChars: MIN_CHARS });
+      default:
+        return noResultsText;
+    }
+  };
+
+  const error = props.errors ?? queryError ?? null;
+
+  if (!visible) return null;
+
+  return (
+    <DetailInputWithLabelRow
+      sx={DefaultFormRowSx}
+      label={label}
+      labelWidthPercentage={FORM_LABEL_WIDTH}
+      inputAlignment={'start'}
+      Input={
+        editMode ? (
+          <Autocomplete
+            sx={{
+              '.MuiFormControl-root': { minWidth: '100%' },
+              flexBasis: '100%',
+            }}
+            options={results}
+            disabled={!props.enabled}
+            onChange={(_, option) => handleDataUpdate(option)}
+            onInputChange={(_, value) => {
+              debouncedOnChange(value);
+              setSearchText(value);
+              setNoResultsText(t('control.search.searching-label'));
+            }}
+            onBlur={() => {
+              if (data) setEditMode(false);
+            }}
+            getOptionLabel={getOptionLabel ?? undefined}
+            clearable={!props.config?.required}
+            inputProps={{
+              error: !!error,
+              helperText: error,
+            }}
+            noOptionsText={getNoOptionsText()}
+            renderInput={params => (
+              <BasicTextInput
+                {...params}
+                placeholder={
+                  placeholderText ?? t('control.search.search-placeholder')
+                }
+              />
+            )}
+          />
+        ) : (
+          <Box
+            display="flex"
+            alignItems="center"
+            justifyContent="space-between"
+            flexBasis="100%"
+            sx={{ width: FORM_INPUT_COLUMN_WIDTH }}
+          >
+            {!error ? (
+              <>
+                {getDisplayElement && getDisplayElement(data)}
+                <IconButton
+                  label={t('label.edit')}
+                  icon={<EditIcon style={{ width: 16 }} />}
+                  onClick={() => {
+                    setEditMode(true);
+                  }}
+                  color="primary"
+                  height="20px"
+                  width="20px"
+                />
+              </>
+            ) : (
+              <Typography color="red">{error}</Typography>
```

Comment:
this is done in four places that I've seen, and should be `color="error"` so that we can control the error colour in the theme rather than hard coding it here.

Reply:
Ok thanks, I fixed it